specification/keyvault/Security.KeyVault.BackupRestore/models.tsp

```diff
@@ -0,0 +1,244 @@
+import "@typespec/http";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "../Security.KeyVault.Common/common.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+
+namespace KeyVault;
+
+/**
+ * An authentication method and location for the operation.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "Model name has already shipped"
+model SASTokenParameter {
+  /**
+   * Azure Blob storage container Uri
+   */
+  storageResourceUri: string;
+
+  /**
+   * The SAS token pointing to an Azure Blob storage container
+   */
+  token?: string;
+
+  /**
+   * Indicates which authentication method should be used. If set to true, Managed HSM will use the configured user-assigned managed identity to authenticate with Azure Storage. Otherwise, a SAS token has to be specified.
+   */
+  useManagedIdentity?: boolean = false;
+}
+
+/**
+ * The status of a long-running operation.
+ */
+@lroStatus
+union OperationStatus {
+  string,
+
+  /** The operation is in progress. */
+  InProgress: "InProgress",
+
+  /** The operation successfully completed. */
+  Succeeded: "Succeeded",
+
+  /** The operation was canceled. */
+  Canceled: "Canceled",
+
+  /** The operation failed. */
+  Failed: "Failed",
+}
+
+/**
+ * Full backup operation
+ */
+model FullBackupOperation {
+  /**
+   * Status of the backup operation.
+   */
+  status?: OperationStatus;
```

Comment:
I am surprised that `status` is not required. I assume that this is because optionality is/was loosy-goosily defined in the source swagger/openapi?

Reply:
Yep, that's correct -- none of the attributes on operation models are required in the original spec, either by oversight or for an obscure edge case

Reply:
Why is status now of type OperationStatus? Looking at the swagger I'm generating from, it's just a string. Would be a breaking change for Go.

Reply:
https://github.com/Azure/azure-rest-api-specs/pull/30239#discussion_r1893416050

Reply:
Just to confirm, I tried changing `status` to a `string` and suppressing the warning related to not having a valid LRO status monitor. The TSP compiles, but the operation no longer registers as an LRO and wouldn't work for generation. The LRO designations for these operations are lost in the emitted Swagger:
<img width="354" alt="Git diff screenshot showing that two LRO-related extensions have been removed from the Swagger" src="https://github.com/user-attachments/assets/d2e8968a-78a3-485e-b52e-452328e7cec7" />